### PR TITLE
Added yard2LCD driver to lcdproc

### PR DESCRIPTION
**Pull request by YARD2**
Added a new driver for yard2LCD hardware controller

### CODE REVIEW

File: server/drivers/yard2LCD.c [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+				return -1;
+			
+			// Setup and send YARD command
+			cmdBuf[0] = 'G';
+			cmdBuf[1] = x - 1;	// Zero index start ! lcdproc starts with 1,1, YARD 0,0
+			cmdBuf[2] = y - 1;
+			return yard_hwWrite(drvthis, cmdBuf, 3);
+			
+			break;
+		}
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:175]: No need for break after return

File: server/drivers/yard2LCD.c [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+	
+	// Allocate framebuf & Setup frame buffer x2 to be sure that the buffer is big enough
+	p->framebuf = (unsigned char *) malloc((p->width * p->height)*2);
+	if (p->framebuf == NULL) 
+	{
+		report(RPT_ERR, "%s: Can't create framebuffer !", drvthis->name);
+		return -1;
+	}
+	memset(p->framebuf, ' ', p->width * p->height);
+
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:356]: Why do you need a double size frame buffer but then only initialize the first half?

File: server/drivers/yard2LCD.c [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+/* not done yet
+	// Init size and graphical size
+*/	
+
+/*	
+	// Init backlight-on brightness -> done by YARD2 normally in auto mode
+*/
+	tmp = DEFAULT_ON_BRIGHTNESS;
+	p->on_brightness = tmp;
```
Comment (haraldg) [inline server/drivers/yard2LCD.c]: What's the point of tmp?

File: server/drivers/yard2LCD.c [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+	PrivateData *p = (PrivateData *)drvthis->private_data;
+	
+	// Reset custom char mode and clear frambuffer
+	p->ccmode = standard;
+	memset(p->framebuf, ' ', p->width * p->height);
+}
+
+/*
+ * Flushes a single frame to the VFD
+ */
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:435]: VFD?

File: server/drivers/yard2LCD.c [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+	x--; //start with 0,0
+	y--;
+	for (i = 0; string[i] != '\0'; i++) 
+	{
+		// Check for buffer overflow
+		if ( (y * p->width + x + i) >= (p->width * p->height) )
+			break;
+		p->framebuf[(y * p->width) + x + i] = string[i];
+	}
+}
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:514]: This function should check if the string actually fits on the line. See:
http://lcdproc.sourceforge.net/docs/current-dev.html#function-details

File: server/drivers/yard2LCD.c [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+ * Provides info about the driver
+ */
+MODULE_EXPORT const char *
+yard_get_info(Driver *drvthis)
+{
+	debug(RPT_DEBUG, "%s: Event 23 - Enter yard_get_info",drvthis->name);
+	PrivateData *p = (PrivateData *)drvthis->private_data;
+	
+	strcpy(p->info, "Socket-based driver for Y.A.R.D.2 USB LCD");
+	return p->info;
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:657]: This looks more like a driver for a YARD2 server then for a USB device ...

File: server/drivers/yard2LCD.c [hunk 7]
```diff
@@ -0,0 +1,647 @@
+/***************************************************************************
+ *   Copyright (C) 2009 by M. Feser                                        *
+ *   Copyright (C) 1013 by R. Geigenberger (Update for YARD2)              *
+ *                                                                         *
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:5]: Please include an e-mail address where you expect to remain reachable. I think lcdproc has already too many drivers where the author can't be contacted easily.

File: server/drivers/yard2LCD.c [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+	int bigcellwidth, bigcellheight;
+	char *framebuf;
+	int on_brightness;
+	int off_brightness;
+	char hw_brightness;
+	CGmode ccmode;
+	Dmode dispmode;
+	char info[255];
+	char LCDtype;
+} PrivateData;
```
Comment (haraldg) [inline server/drivers/yard2LCD.c:74]: A lot of these properties (like cellwidth) aren't used at all except for shuffling defaults back and forth. Unless you are planning to actually implement this, consider omitting this. Especially get rid of it, if the protocoll of your server doesn't support it anyway.

**Comment by haraldg** [comment]
Thanks for your code. I made a number of points inline. On the global scale please have a look at the coding style:
http://lcdproc.sourceforge.net/docs/current-dev.html#code-style
and add proper documentation about your driver:
http://lcdproc.sourceforge.net/docs/current-dev.html#documentation

File: server/drivers/yard2LCD.c (hunk 1 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:175]: Fixed

File: server/drivers/yard2LCD.c (hunk 2 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:356]: Fixed, init complete FB

File: server/drivers/yard2LCD.c (hunk 3 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c]: removed tmp

File: server/drivers/yard2LCD.c (hunk 4 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:435]: Changed to LCD

File: server/drivers/yard2LCD.c (hunk 5 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:514]: Hopefully fixed

File: server/drivers/yard2LCD.c (hunk 6 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:657]: Yes the main program is a background server, yard2srvd, which handles the LCD output

File: server/drivers/yard2LCD.c (hunk 7 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:5]: Added Mail

File: server/drivers/yard2LCD.c (hunk 8 above)
Comment (YARD2) [inline server/drivers/yard2LCD.c:74]: For later I plan to implement graphical LCD support

**Comment by willfe** [comment]
Closing (superseded by #19).

**Pull request closed without merging**